controlplane/kubeadm/api/v1alpha4/kubeadm_control_plane_webhook.go

```diff
@@ -101,6 +105,10 @@ func (in *KubeadmControlPlane) ValidateUpdate(old runtime.Object) error {
 		{spec, kubeadmConfigSpec, clusterConfiguration, "dns", "imageRepository"},
 		{spec, kubeadmConfigSpec, clusterConfiguration, "dns", "imageTag"},
 		{spec, kubeadmConfigSpec, clusterConfiguration, "imageRepository"},
+		{spec, kubeadmConfigSpec, clusterConfiguration, apiServer, "*"},
+		{spec, kubeadmConfigSpec, clusterConfiguration, controllerManager, "*"},
+		{spec, kubeadmConfigSpec, clusterConfiguration, scheduler, "*"},
+		{spec, kubeadmConfigSpec, clusterConfiguration, featureGates, "*"},
```

Comment:
I'm a little bit concerned about allowing mutability of this field, because there are no guarantee that the impact of a change here is limited to the control plane components/args.

For instance, as of today, there is a feature flag that allows changing the algorithm to be used for generating certificates keys, and nothing prevents kubeadm maintainers to add a feature gate flag impacting other bits which can cause problems in the context of an immutable rollout.

What about keeping this immutable unless there is a specific use case for this?

Reply:
I'm not aware of any immediate use case for kubeadm feature flags. I'm okay with reverting it to make it immutable. @ncdc, wdyt?

Reply:
It felt like a potentially reasonable thing to open up (or, conversely, that keeping it immutable was potentially unnecessarily restrictive). If someone wants to roll out changes that require modifying feature gate flags in both the control plane and kubelets, unless we make this field mutable, that use case would not be allowed?

Reply:
iiuc, this only controls the feature gates for kubeadm. Right now, there are only two feature gates in kubeadm (PublicKeysECDSA, IPv6DualStack). For other kubernetes components, we'll need to use `extraArgs` and set the featureGates there.

Reply:
Ohhhhh thank you for that clarification! I'll defer to @fabriziopandini in that case - would someone ever want to adjust which kubeadm feature gates are in use through a rolling upgrade?

Reply:
I'm for keeping this immutable.

Reply:
I've reverted featureGates mutation commit, PTAL.